### ISSUE TITLE
Always render AddonsByAuthorsCard

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-frontend#6775]
### Describe the problem and steps to reproduce it:

1. go to https://addons-dev.allizom.org/en-US/firefox/user/abine/
2. refresh
3. observe

### What happened?

The left card is rendered but the right side is empty. Then the right side starts to render some stuff, then it is fully loaded with the add-ons by the user.

Current state in -dev:

![2018-10-29 21 03 50](https://user-images.githubusercontent.com/217628/47676951-31c40800-dbbe-11e8-9a12-0ec31a90b083.gif)

### What did you expect to happen?

The full layout should be rendered in a "loading" state.

Expected state locally (hence the no-css render at the beginning):

![2018-10-29 21 06 49](https://user-images.githubusercontent.com/217628/47677152-a72fd880-dbbe-11e8-9d82-c2ad980584be.gif)

**Comment by AlexandraMoga** [comment]
@willdurand sorry, but I don't find the current implementation to be much better than before.
The Extensions card expands too much and the loading state appears even if the user doesn't have any extensions listed. The same goes for the Themes card.  
There is also no loading state rendered for the User reviews card - it disappears shortly at refresh. 
I've checked prod for comparison and this behavior exists there as well, but is not that noticeable. 

The Extensions and Themes cards are not disappearing anymore though :)
I should probably file two separate issues for what I mentioned above. 

This is how the new implementation looks to me on stage:

![author cards](https://user-images.githubusercontent.com/31961530/47785142-795aa900-dd10-11e8-84ed-24de1fa1f70d.gif)

**Comment by willdurand** [comment]
> @willdurand sorry, but I don't find the current implementation to be much better than before.
> The Extensions card expands too much and the loading state appears even if the user doesn't have any extensions listed. The same goes for the Themes card.
> There is also no loading state rendered for the User reviews card - it disappears shortly at refresh.
> I've checked prod for comparison and this behavior exists there as well, but is not that noticeable.
> 
> The Extensions and Themes cards are not disappearing anymore though :)
> I should probably file two separate issues for what I mentioned above.

@AlexandraMoga would you have a comparison with -prod? I am not sure I understand. It should be slightly better, not completely different 😛 

The user reviews issue is tracked in https://github.com/mozilla/addons-frontend/issues/6309.

**Comment by AlexandraMoga** [comment]
@willdurand sure, here is an example from prod:

![author cards prod](https://user-images.githubusercontent.com/31961530/47836448-02c1b800-ddb1-11e8-9b2b-04155aac681b.gif)

**Comment by bobsilverberg** [comment]
FWIW, there was a [talk at ReactConf](https://www.youtube.com/watch?v=ByBPyMBTzM0) about just this type of thing: not showing loading spinners and content for very short periods of time. It's not exactly the same thing here, because the problem here is exacerbated by the fact that the loading indicators are so large and that they occupy space that is actually entirely removed once the loading is finished, but the idea is the same: Loading spinners/content can actually degrade the user experience.

The talk was about the new React `suspense` API, which is available in React 16.6 (although apparently doesn't play well with SSR), and the argument presented was that it's better to show nothing than a spinner/loading gif, if the content is not going to take very long to load.

The suspense API is designed to help with that issue of how long it might take the content to load, but in this case it does seem like maybe we'd be better off showing nothing until the content loads rather than the loading indicators because it is possible (or maybe even likely) that nothing at all will load.

**Comment by kumar303** [comment]
Ah, yes, I reviewed this patch but now I realize that a delayed loading state would be better since extensions, themes, and reviews might not ever appear on the user profile page.

First, the ReactConf talk up above is talking about showing spinners for a loading state which is a different UI strategy than ours. We show skeleton text. It's better to show skeleton text immediately because it creates the illusion that the page is immediately ready (and makes the site feel snappy). However, showing immediate skeleton text is *only* helpful for layouts where we know for sure that text will fill in the placeholders. For layouts where the text might not appear, we should consider a delayed loading state.

For example; when the user loads an add-on page, it makes sense to show immediate skeleton text for the description because we know a description will appear. It does not make sense to show immediate skeleton text for add-ons by authors since we are not sure that this text will appear or not. For fast-ish connections, the add-ons by authors data will appear when ready. For slower connections, we may want to show a loading state to indicate that the page is not yet completely laid out.

@willdurand what do you think about changing it to a delayed loading state? (In a new issue, of course.)

**Comment by willdurand** [comment]
I have to think more about it. Once URLs have user IDs instead of usernames (Q4), we could get all the data on the server. That'd be ideal because no need to render loading indicators.

Maybe we should not hide the cards and add a message saying there is no extensions or themes. Because if viewing the loading state and then nothing is weird, I'd argue that not viewing anything and then getting the cards displayed magically is also odd.
